### PR TITLE
Loading of poptype needs & rebel compositions

### PR DESCRIPTION
**Pull request by k0uneli**

### CODE REVIEW

File: src/openvic-simulation/dataloader/Dataloader.cpp [hunk 1]
```diff
@@ -298,7 +298,11 @@ bool Dataloader::load_defines(GameManager& game_manager) const {
 		Logger::error("Failed to load goods!");
 		ret = false;
 	}
-	if (!_load_pop_types(game_manager.get_pop_manager(), pop_type_directory)) {
+	if (!_load_units(game_manager, units_directory)) {
+		Logger::error("Failed to load units!");
+		ret = false;
+	}
+	if (!_load_pop_types(game_manager.get_pop_manager(), game_manager, pop_type_directory)) {
```
Comment (Spartan322) [inline src/openvic-simulation/dataloader/Dataloader.cpp:305]: Probably be better to leave reorganizing the position of `_load_units` to another PR, (a lot of reorganization is gonna be needed soon anyway, like running formatting and include-what-you-use) honestly we should really start moving to a "minimal purpose built changes" model for PRs, but that can be discussed in dev chat, for now the `_load_units` move should be moved to a separate PR.

**Comment by Hop311** [comment]
Good work, I'm gonna leave merging until after some other PRs are merged + I have the chance to rebase this branch to include those changes.

File: src/openvic-simulation/dataloader/Dataloader.cpp (hunk 1 above)
Comment (Hop311) [reply src/openvic-simulation/dataloader/Dataloader.cpp:305]: The reorganisation isn't arbitrary/separate to the main purpose of the PR, it's necessary to be able to look up unit types while parsing pop types.

**Review by Hop311 — APPROVED** [review]
